### PR TITLE
Remove scheduler "TestGroup" utility.

### PR DESCRIPTION
**Pull request by misterikkit**
**What this PR does / why we need it**:

This util was used to fake certain aspects of apiserver behavior, such
as resource paths and JSON encoding. Our unit tests have been refactored
so they don't rely on the REST or JSON aspects of apiserver. This util
is no longer needed.

**What type of PR is this?**
/kind cleanup
/sig scheduling
/cc @kubernetes/sig-api-machinery-pr-reviews

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
I'm not sure if testing different encoders based on the `KUBE_TEST_API_TYPE` variable is at all relevant to scheduler tests. I expect it isn't.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/factory/factory_test.go [hunk 1]
```diff
@@ -366,18 +365,15 @@ func testBind(binding *v1.Binding, t *testing.T) {
 
 	pod := client.CoreV1().Pods(metav1.NamespaceDefault).(*fakeV1.FakePods)
 
-	bind, err := pod.GetBinding(binding.GetName())
+	actualBinding, err := pod.GetBinding(binding.GetName())
```
Comment (yastij) [inline pkg/scheduler/factory/factory_test.go]: +1 the naming wasn't clear on this one

**Review by yastij — COMMENTED** [review]
This mostly lgtm (just make the CI happy), thanks @misterikkit for the refactoring !

**Review by Huang-Wei — APPROVED** [review]
/lgtm

Thanks @misterikkit.

**Review by bsalamat — COMMENTED** [review]
/approve

**Pull request merged**